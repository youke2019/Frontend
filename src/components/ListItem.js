import React from "react";
import {
    Text,
    StyleSheet,
    View,
    Image,
    TouchableOpacity
} from "react-native";

class ListItem extends React.Component {
    render(){
        const {
            text= '',
            image= '',
            onPress= () => {},
        } = this.props

        return (
            <View style={styles.container}>
                <View style={styles.text_container}>
                    <Image
                        style={styles.text_image}
                        source={{uri:image}}
                    />
                    <Text style={styles.text}>{text}</Text>
                </View>
                <TouchableOpacity
                    style={styles.arrow_container}
                    onPress={onPress}
                >
                    <Image
                        source={{uri:'right_arrow'}}
                        style={styles.arrow_image}
                    />
                </TouchableOpacity>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container:{
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingVertical: 10,
        paddingHorizontal: 10,
        paddingTop: 20,
        paddingBottom: 8
    },
    text_container:{
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        paddingHorizontal: 5,
    },
    text_image:{
        width:22,
        height:22,
    },
    text:{
        paddingHorizontal:8,
        fontSize: 16,
    },
    arrow_container:{
        alignItems: 'center',
        justifyContent: 'center'
    },
    arrow_image:{
        width:20,
        height:20,
    },
    divider:{
        height:1
    },
})

export default ListItem