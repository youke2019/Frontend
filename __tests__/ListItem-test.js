import React from 'react'
import ListItem from '../src/components/ListItem'
import Adapter from 'enzyme-adapter-react-16'
import {shallow,configure} from 'enzyme'

configure({adapter: new Adapter()})

import renderer from 'react-test-renderer';

test('render an empty ListItem', () => {
    const tree = renderer.create(<ListItem />).toJSON();
    expect(tree).toMatchSnapshot();
});

const text = '课程'
const image = 'nop'
const onPress = jest.fn(() => {

})

test('render ListItem with data', () => {
    let tree = renderer.create(
        <ListItem
            text={text}
            image={image}
            onPress={onPress}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

test('test onPress', () => {
    let wrapper = shallow(
        <ListItem
            text={text}
            image={image}
            onPress={onPress}
        />
    )

    wrapper.find('TouchableOpacity').props().onPress()
    expect(onPress).toHaveBeenCalled();
});


