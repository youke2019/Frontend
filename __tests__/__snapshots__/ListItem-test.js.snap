// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ListItem with data 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingHorizontal": 10,
      "paddingTop": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "nop",
        }
      }
      style={
        Object {
          "height": 22,
          "width": 22,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 16,
          "paddingHorizontal": 8,
        }
      }
    >
      课程
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "right_arrow",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`render an empty ListItem 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingBottom": 8,
      "paddingHorizontal": 10,
      "paddingTop": 20,
      "paddingVertical": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "height": 22,
          "width": 22,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 16,
          "paddingHorizontal": 8,
        }
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "right_arrow",
        }
      }
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
